### PR TITLE
An use vector for rotation

### PR DESCRIPTION
**Pull request by AndreaNeumayr**
Changed Quaternion to SVector{4,F} and RotationMatrix to SMatrix{3,3,F,9}

### CODE REVIEW

File: src/Composition/frameMeasurements.jl [hunk 1]
```diff
@@ -1,37 +1,37 @@
 """
     R = measFrameRotation(frameMeas::Object3D; frameOrig::Object3D)
 
-Return relative RotationMatrix `R` from frame `frameOrig` into frame `frameMeas`.
+Return relative SMatrix `R` from frame `frameOrig` into frame `frameMeas`.
```
Comment (GerhardHippmann) [inline src/Composition/frameMeasurements.jl:4]: In docu "rotation matrix" would be clearer.

File: src/Composition/object3D.jl [hunk 2]
```diff
@@ -178,7 +178,8 @@ mutable struct Object3D{F} <: Modia3D.AbstractObject3D
 
         if !isnothing(parent)
             # with parent -> call constructor 3
-            if !isnothing(rotation) && (typeof(rotation) != Frames.RotationMatrix)
+            if !isnothing(rotation)
+                rotation = Modia3D.convertAndStripUnit(SVector{3,F}, u"m"  , rotation)
```
Comment (GerhardHippmann) [inline src/Composition/object3D.jl:182]: "m" is wrong for angles, should be "rad"

File: src/Frames/_module.jl [hunk 3]
```diff
@@ -33,7 +33,7 @@ number (interpreted as having unit `rad`) or a number with a unit
 
 # Constructors for a RotationMatrix R
 
-The following functions return a [`Modia3D.RotationMatrix`](@ref)` R`
+The following functions return a `SMatrix{3,3,F,9}` R`
```
Comment (GerhardHippmann) [inline src/Frames/_module.jl:36]: Please add "rotation matrix"

File: src/Frames/quaternion.jl [hunk 4]
```diff
@@ -7,34 +7,32 @@
...
-NullQuaternion = Quaternion(0.0, 0.0, 0.0, 1.0)
+NullQuaternion(::Type{F}) where {F} = SVector{4,F}(0.0, 0.0, 0.0, 1.0)
+
 
 
 """
     Modia3D.assertQuaternion(q::AbstractVector)
 
-Assert that vector `q` has the properties of a `Quaternion` vector
+Assert that vector `q` has the properties of a `SVector{4,F}` vector
```
Comment (GerhardHippmann) [inline src/Frames/quaternion.jl:35]: Please add "quaternion"

File: src/Frames/rotationMatrix.jl [hunk 5]
```diff
@@ -9,31 +9,7 @@
...
-v1 = R'*v2
-```
-"""
-const RotationMatrix = SMatrix{3,3,Float64,9}
-# RotationMatrix(F) = SMatrix{3,3,F,9}
-
-
-"""
-Constant RotationMatrix that defines no rotation from frame 1 to frame 2.
+Constant SMatrix that defines no rotation from frame 1 to frame 2.
```
Comment (GerhardHippmann) [inline src/Frames/rotationMatrix.jl:12]: Please add "rotation matrix"

File: test/Robot/YouBotsGripping.jl [hunk 6]
```diff
@@ -559,7 +559,7 @@ stopTime = 28.0
...
+        requiredFinalStates = [0.0050970438521427686, 0.7904209132350711, 0.18419404940475442, -1.8137251625912176e-16, 5.15723222832311e-16, 4.3563251639972627e-16, -1.4095490501776007, 1.052597940983404, 0.3609484049540662, -3.643852247439651e-15, 2.7004689815539263e-16, -1.898293154574278e-14, -5.035685853449397e-13, 5.038187563005594e-13, -6.423546330581647e-12, 6.424367473457469e-12, -1.466646015541777e-11, 1.46683460703095e-11, -9.012171586569444e-12, 9.013600853782148e-12, 9.032215135499194e-14, -9.032784745301517e-14, -3.847327194783694e-9, 0.24163532322220996, -0.0007949230980801731, -0.0009396161969942047, 3.1421054610143916e-10, -1.0373829067802092e-8, 0.0009999997883376663, 2.1161485005970094e-10, 4.7610113997790455e-7, -4.758629591336826e-7, -4.541758289187212e-6, 4.54228619794064e-6, -8.529188503662272e-6, 8.530367912683205e-6, -5.053575394379844e-6, 5.0544367215036136e-6, 3.5345587109034675e-8, -3.5349028754053165e-8, 0.003640521453754823, 0.20692212203816962, -0.042582089475276606, -0.018517919284936582, 0.00012295508999650139, -0.001425191900008411, 0.0009709211088039017, 2.907249543964317e-5]
```
Comment (GerhardHippmann) [inline test/Robot/YouBotsGripping.jl:562]: Why do results change?

Comment (AndreaNeumayr) [reply test/Robot/YouBotsGripping.jl:562]: I don't know. I changed from `rot1, rot3, rot123` to vectors like I did for the other robot models (the same values are used there). 
I ran this model several times. New final states are required only if I change `rot3` to vector in line 430.

File: src/Composition/object3D.jl (hunk 2 above)
Comment (AndreaNeumayr) [reply src/Composition/object3D.jl:182]: Thanks.

File: test/Robot/YouBotsGripping.jl (hunk 6 above)
Comment (GerhardHippmann) [reply test/Robot/YouBotsGripping.jl:562]: I checked this:

```
julia> Modia3D.rot3(-180u"°")
3×3 SMatrix{3, 3, Float64, 9} with indices SOneTo(3)×SOneTo(3):
 -1.0  -0.0  0.0
  0.0  -1.0  0.0
  0.0   0.0  1.0

julia> Modia3D.rot3(-pi)
3×3 SMatrix{3, 3, Float64, 9} with indices SOneTo(3)×SOneTo(3):
 -1.0          -1.22465e-16  0.0
  1.22465e-16  -1.0          0.0
  0.0           0.0          1.0

```

**Pull request merged**